File: server/routes/wishlistRoutes.js
const express = require('express');
const router = express.Router();
const { wishlistController } = require('../controllers');

// API to add an item to the wishlist new update now
router.post('/add', wishlistController.addToWishlist);

// API to update an item in the wishlist
router.put('/update/:job_id', wishlistController.updateWishlistItem);

// API to delete an item from the wishlist 
router.delete('/delete', wishlistController.deleteFromWishlist);

// API to check if a job_id exists in the wishlist 
router.get('/check', wishlistController.checkJobIdInWishlist);

// API to get all wishlist items for a specific user
router.get('/all', wishlistController.getAllWishlistItems);

module.exports = router;

File: server/router/wishListRoutes.go
// wishlist
package routes

import (
	"Ezhire/controllers"

	"github.com/gorilla/mux"
)

// WishlistRoutes sets up the routes for wishlist management
func WishlistRoutes(router *mux.Router) {
	router.HandleFunc("/add", controllers.AddToWishlist).Methods("POST")
	router.HandleFunc("/update/{job_id}", controllers.UpdateWishlistItem).Methods("PUT")
	router.HandleFunc("/delete", controllers.DeleteFromWishlist).Methods("DELETE")
	router.HandleFunc("/check", controllers.CheckJobIdInWishlist).Methods("GET")
	router.HandleFunc("/all", controllers.GetAllWishlistItems).Methods("GET")
}
